diff --git a/src/OpenFOAM/primitives/Scalar/longDoubleScalar/longDoubleScalar.H b/src/OpenFOAM/primitives/Scalar/longDoubleScalar/longDoubleScalar.H
index b886589722..787fd706c4 100644
--- a/src/OpenFOAM/primitives/Scalar/longDoubleScalar/longDoubleScalar.H
+++ b/src/OpenFOAM/primitives/Scalar/longDoubleScalar/longDoubleScalar.H
@@ -151,12 +151,26 @@ inline Scalar atan2(const Scalar y, const Scalar x)
 
 inline Scalar jn(const int n, const Scalar s)
 {
+#ifdef freebsd64
+    return static_cast<Scalar>
+    (
+        ::jn(n, static_cast<double>(s))
+    );
+#else
     return ::jnl(n, s);
+#endif
 }
 
 inline Scalar yn(const int n, const Scalar s)
 {
+#ifdef freebsd64
+    return static_cast<Scalar>
+    (
+        ::yn(n, static_cast<double>(s))
+    );
+#else
     return ::ynl(n, s);
+#endif
 }
 
 #undef Scalar
