diff --git a/src/OpenFOAM/global/fileOperations/masterUncollatedFileOperation/masterUncollatedFileOperation.C b/src/OpenFOAM/global/fileOperations/masterUncollatedFileOperation/masterUncollatedFileOperation.C
index c278b4706e..b2d2dc0a9f 100644
--- a/src/OpenFOAM/global/fileOperations/masterUncollatedFileOperation/masterUncollatedFileOperation.C
+++ b/src/OpenFOAM/global/fileOperations/masterUncollatedFileOperation/masterUncollatedFileOperation.C
@@ -848,7 +848,7 @@ bool Foam::fileOperations::masterUncollatedFileOperation::mkDir
     mode_t mode
 ) const
 {
-    return masterOp<mode_t, mkDirOp>
+    return masterOp<uint32_t, mkDirOp>
     (
         dir,
         mkDirOp(mode),
@@ -864,7 +864,7 @@ bool Foam::fileOperations::masterUncollatedFileOperation::chMod
     mode_t mode
 ) const
 {
-    return masterOp<mode_t, chModOp>
+    return masterOp<uint32_t, chModOp>
     (
         fName,
         chModOp(mode),
@@ -881,12 +881,15 @@ mode_t Foam::fileOperations::masterUncollatedFileOperation::mode
     const bool followLink
 ) const
 {
-    return masterOp<mode_t, modeOp>
+    return static_cast<mode_t>
     (
-        fName,
-        modeOp(checkVariants, followLink),
-        Pstream::msgType(),
-        comm_
+        masterOp<uint32_t, modeOp>
+        (
+            fName,
+            modeOp(checkVariants, followLink),
+            Pstream::msgType(),
+            comm_
+        )
     );
 }
 
